learn/experimental/vector_search.mdx

```diff
@@ -30,39 +30,151 @@ curl \
 </Capsule>
 
-### Add embeddings to Meilisearch
+#### Generate auto-embeddings with HuggingFace 
+
+The HuggingFace computes embeddings locally. This is a resource-intensive operation and might affect indexing performance.
+
+Use the `embedders` index setting of the [update `/settings` endpoint](/reference/api/settings) to configure one or more embedders for an index:
+
+```sh
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "embedders": {
+      "default": {
+        "source": "huggingFace",
+        "model": "bge-base-en-v1.5",
+        "documentTemplate": "A movie titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"
+      }
+    }
+  }'
+```
+
+`documentTemplate` is an optional field you can use to customize the data you send to the embedder. It is highly recommended you configure a custom template for your documents.
+
+<Capsule intent="tip" title="`documentTemplate` usage">
+`documentTemplate` must be a [Liquid template](https://shopify.github.io/liquid/). Use `{{ doc.attribute }}` to access document field values. Meilisearch also exposes a `{{ fields }}` array containing one object per document field, which you may access with `{{ field.name }}` and `{{ field.value }}`.
+
+For best results, use short strings indicating the type of document in that index, only include highly relevant document fields, and truncate long fields.  
+</Capsule>
+
+#### Custom embeddings
+
+You may also provide custom embeddings. In this case, you must manually update your embeddings when adding, updating, and removing documents to your index.
 
-Use [the `/documents` endpoint](/reference/api/documents) to upload vectorized documents. Store vector data in your documents' `_vector` field:
+Configure the `embedder` index setting:
+
+```sh
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "embedders": {
+      "image2text": {
+        "source":  "userProvided",
+        "dimensions": 3
+      }
+    }
+  }'
+```
+
+Then, use [the `/documents` endpoint](/reference/api/documents) to upload vectorized documents. Store vector data in your documents' `_vectors` field:
 
 ```sh
 curl -X POST -H 'content-type: application/json' \
 'localhost:7700/indexes/products/documents' \
 --data-binary '[
-    { "id": 0, "_vectors": [0, 0.8, -0.2], "name": "frying pan" },
-    { "id": 1, "_vectors": [1, -0.2, 0], "text": "baking dish" },
-    { "id": 2, "_vectors": [[0.5, 3, 1], [-0.2, 4, 6]], "text": "spork" }
+    { "id": 0, "_vectors": {"image2text": [0, 0.8, -0.2]}, "text": "frying pan" },
+    { "id": 1, "_vectors": {"image2text": [1, -0.2, 0]}, "text": "baking dish" }
 ]'
 ```
 
-### Use `vector` to query Meilisearch
+### Vector search with auto-embeddings
```

Comment:
```suggestion
### Hybrid search with auto-embeddings
```

Reply:
For the moment, while our communication strategy for this hasn't been completely decided by the marketing team, I will keep vector search so it's consistent with the rest of this document. Once we have clarity on terminology choices and etc, I'll update this article to reflect that.